app/src/main/java/com/graeberj/taskman/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,65 @@
+package com.graeberj.taskman.agenda.data
+
+import com.graeberj.taskman.agenda.data.model.Agenda
+import com.graeberj.taskman.agenda.data.model.AgendaItem
+import com.graeberj.taskman.agenda.data.model.Attendee
+import com.graeberj.taskman.agenda.data.remote.api.AgendaApi
+import com.graeberj.taskman.agenda.domain.repository.AgendaRepository
+import kotlinx.coroutines.flow.Flow
+import java.time.LocalDate
+
+class AgendaRepositoryImpl (
+    private val api: AgendaApi
+) : AgendaRepository{
+    override suspend fun insertReminder(reminder: AgendaItem.Reminder, isEdit: Boolean) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun getReminderById(id: String): AgendaItem.Reminder {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun deleteReminderById(id: String) {
+        TODO("Not yet implemented")
+    }
+
+    override fun getAgenda(date: LocalDate, forceRemote: Boolean): Flow<Agenda> {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun insertTask(task: AgendaItem.Task, isEdit: Boolean) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun changeStatusTask(id: String, isDone: Boolean) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun getTaskById(id: String): AgendaItem.Task {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun deleteTaskById(id: String) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun getAttendee(email: String): Result<Attendee?> {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun getEventById(id: String): AgendaItem.Event {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun insertEvent(event: AgendaItem.Event, isEdit: Boolean) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun deleteEventById(id: String) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun getAllUpcomingItems(): Agenda {
+        TODO("Not yet implemented")
+    }
+}
```

Comment:
I'd keep this a bit separate based on responsibilities. You could split this rather large repo into `EventRepository`, `TaskRepository` and `ReminderRepository`. That way, each detail screen also only gets the functionality it needs. The agenda screen can get all those repos and combine their result